### DISCUSSION TITLE
What are the potential reasons for high CPU usage spikes?

### DISCUSSION THREAD

**Discussion by karthikvrao** [vercel/next.js#30662] [Help]
We have our production Next.js (v.10.2.3) app running on 5 pods via Kubernetes. Every day after a few hours, the CPU usage starts to spike and reaches 100%. When this happens the app becomes unresponsive and even a request to a login page takes eons.
![image](https://user-images.githubusercontent.com/20852493/139530213-3ec2b9a1-055d-4471-ad16-e38738106f26.png)

The only option we have had was to kill the pods and spawn new pods. The new pods also start off with low CPU usage and end up same way after few hours.
![image](https://user-images.githubusercontent.com/20852493/139530221-de5c3425-21e9-4955-bcb4-bfa5e1e943a1.png)

We checked our code but couldn't find any obvious reasons for why this could be happening.

Any suggestions pointing to potential reasons for this to happen or help on how to fix this issue is greatly appreciated!

**Comment by dosmond** [comment]
Was any solution found on this? We are experiencing similar issues deploying Nextjs to an Azure App Service

**Reply by mihir-karbelkar-wiz** [reply]
Same here, any input would be helpful.

**Reply by harkins-it** [reply]
Same here
